common/network-common/src/main/java/org/apache/spark/network/util/NettyOutOfMemoryError.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.util;
+
+/**
+ * This class is for test only. It's needed because neither the constructor of Netty's
+ * {@link io.netty.util.internal.OutOfDirectMemoryError} can be accessed nor be mocked
+ * due to the type is declared as final.
+ */
+public class NettyOutOfMemoryError extends OutOfMemoryError {
```

Comment:
Nit: Rename to make it clear this is just for test ?

Thought: Can we use `SparkOutOfMemoryError` instead ? We can move it out to a module accessible from here as well ?

Reply:
> Nit: Rename to make it clear this is just for test ?

Yes, sure.

> Thought: Can we use SparkOutOfMemoryError instead ? We can move it out to a module accessible from here as well ?

I think we can't. `SparkOutOfMemoryError` indicates to be a real memory issue of Spark. So we should abort the Spark application instead of catching it when it throws. However, Netty OOM is a different case. As I mentioned in PR desc, Netty manages its own memory region so that Spark is ok to catch it.

Reply:
Sounds good